clients/src/main/java/org/apache/kafka/common/serialization/Serdes.java

```diff
@@ -111,6 +117,10 @@ public ByteArraySerde() {
             return (Serde<T>) Long();
         }
 
+        if (Float.class.isAssignableFrom(type)) {
```

Comment:
Also, note that `type` is a `Class<T>` -- a check with `instanceof` would not work because it requires an _instance_ of a class, whereas `serdeFrom(Class<T> type)` provides us with class object.